plugins/inputs/intel_powerstat/intel_powerstat.go

```diff
@@ -0,0 +1,493 @@
+// +build linux
+
+package intel_powerstat
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/intel_powerstat/data"
+	"github.com/influxdata/telegraf/plugins/inputs/intel_powerstat/services"
+)
+
+const (
+	cpuC1                = "cpu_c1_state_residency"
+	cpuC6                = "cpu_c6_state_residency"
+	cpuBusyCycles        = "cpu_busy_cycles"
+	cpuBusyFrequency     = "cpu_busy_frequency"
+	cpuFrequency         = "cpu_frequency"
+	cpuTemperature       = "cpu_temperature"
+	percentageMultiplier = 100
+)
+
+var (
+	allowedProcessorModelsForC1C6 = []string{"0x37", "0x4D", "0x5C", "0x5F", "0x7A", "0x4C", "0x86", "0x96", "0x9C",
+		"0x1A", "0x1E", "0x1F", "0x2E", "0x25", "0x2C", "0x2F", "0x2A", "0x2D", "0x3A", "0x3E", "0x4E", "0x5E", "0x55", "0x8E",
+		"0x9E", "0x6A", "0x6C", "0x7D", "0x7E", "0x9D", "0x3C", "0x3F", "0x45", "0x46", "0x3D", "0x47", "0x4F", "0x56",
+		"0x66", "0x57", "0x85", "0xA5", "0xA6", "0x8F", "0x8C", "0x8D"}
+)
```

Comment:
Maybe declare this in `verifyProcessor()` as this is the only user...

Reply:
Oh and make them integers right from the start as you don't use them as string anywhere. This allows to get rid of ` services.ConvertHexArrayToIntegerArray()`.

Reply:
Great point, fixed.